app/Http/Controllers/CustomerController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Customer;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Http\Request;
+use Illuminate\Contracts\View\View;
+
+class CustomerController extends Controller
+{
+    public function showOne($id): View|string
+    {
+        $customer = Customer::where('id', $id)->with('addresses')->get()->first();
+        return ($customer == null) ? "Customer nof found!" : view('customer')->with('customer', $customer);
+    }
+
+    public function showFiltered(Request $request): View
+    {
+        $filter = [
+            'first_name' => explode(' ', $request->input('name') . ' ')[0],
+            'last_name' => explode(' ', $request->get('name') . ' ')[1],
+            'email' => $request->input('email'),
+            'number' => $request->input('number'),
+            'blocked' => $request->input('blocked')
+        ];
+        if ($filter['blocked'] == 'all') {
```

Comment:
Можно было обойтись без копипаста одного и того же кода же формирование запроса:
```suggestion
            $query = Customer::where('first_name', 'like', "%{$filter['first_name']}%")
                    ->where('last_name', 'like', "%{$filter['last_name']}%")
                    ->where('email', 'like', "%{$filter['email']}%")
                    ->where('number', 'like', "%{$filter['number']}%");
            if ($filter['blocked'] == 'all') {
                $query->where('blocked', ($filter['blocked'] == 'blocked') ? true : false);
            }
            $filtered = $query->paginate(10);
```